src/surface.lisp

```diff
@@ -309,6 +309,31 @@ Otherwise, return the copy of the image data along with the pointer."
   (with-cairo-object (surface pointer)
 	(cairo_image_surface_get_stride pointer)))
 
+(defmacro with-surface ((surface &optional surface-name) &body body)
+  (let ((var-name (or surface-name '*surface*)))
+    `(let ((,var-name ,surface))
+       (unwind-protect
+	    (progn ,@body)
+       
+	 (progn (surface-finish ,var-name)
+		(destroy ,var-name))))))
+
```

Comment:
I agree. I must have defvared *surface* somewhere in cl-pango. I also wanted to discuss with-context. It seems like the context should be destroyed at the end. Whatever the case, with-context and with-surface should work the same to limit surprises.